File: binary_search.py
def binarySearch(arr, target):
    length = len(arr)
    if length == 0:
        return False
    
    midIdx = int(length / 2)
    left = arr[0:midIdx]
    right = arr[midIdx + 1:]

    if arr[midIdx] > target:
        return binarySearch(left, target)
    elif arr[midIdx] < target:
        return binarySearch(right, target)
    else:
        return True

li = [-4, 0, 1, 12, 14, 19, 21, 24, 30, 38, 51]

print(binarySearch(li, 12)) # True
print(binarySearch(li, 30)) # True
print(binarySearch(li, 3)) # False
print(binarySearch(li, 28)) # False

File: binary_search.js
function binarySearch(array, target) {
    if (array.length === 0) {
        return false;
    }

    let midIdx = Math.floor(array.length / 2);
    let leftHalf = array.slice(0, midIdx);
    let rightHalf = array.slice(midIdx + 1);

    if(target < array[midIdx]) return binarySearch(leftHalf, target)
    else if(target > array[midIdx]) return binarySearch(rightHalf, target)
    else return true
}

li = [-4, 0, 1, 12, 14, 19, 21, 24, 30, 38, 51]

console.log(binarySearch(li, 12)) // True
console.log(binarySearch(li, 30)) // True
console.log(binarySearch(li, 3)) // False
console.log(binarySearch(li, 28)) // False